### DISCUSSION TITLE
Help with config on generic ESP32.  Trying to make single RGB led appear as 3 distinct leds.

### DISCUSSION THREAD

**Discussion by mattlward** [arendst/Tasmota#21069] [Q&A] [answered]
I have always run Tasmota on devices that had templates that I could add.  This time I do not.  
I am trying to make the first baby steps of an aquarium monitor.  I have the SHT30 working as it should.  But, I am really stuck and struggling to add a single RGB discrete LED.  I am shooting for control of R, G and B to indicate temperature.  I will be running it from Home Assistant.

Any hints to what I am doing wrong?
I have tried the following, this to me looks like 3 individual PWM channels, but only 1 switch appears:
![image](https://github.com/arendst/Tasmota/assets/8540074/2d6dd8d0-df8e-4dc5-911f-dd8093a1b95b)
![image](https://github.com/arendst/Tasmota/assets/8540074/a955d4e0-0a77-4b35-bb97-8b2ceafc96be)

To me this looks like it should be wrong as it would indicate that all three channels are in the same bulb:
![image](https://github.com/arendst/Tasmota/assets/8540074/22f7f61f-052a-4f7c-84d8-79f92c63e45d)
![image](https://github.com/arendst/Tasmota/assets/8540074/02b04307-cdc6-4691-8fe9-6ffddf284aa6)

**Comment by sfromis** [comment, marked as answer]
What you get by 3 PWM channels is that Tasmota will be seeing this as a single lamp, where you can control RGB color mix by selecting nuance and saturation. If you enable `SetOption68`, this will be split into separate channels, where you can regulate brightness for each with the Channel command.

You could also use `SetOption15` to choose basic PWM control, where you can instead use the PWM command in a similar way, without sliders/toggles in the web UI.

The simplest approach, for on/off by channel, is to config the 3 channels as relays, but that would of course not provide brightness.

**Reply by mattlward** [reply]
So, thanks a bunch.  I gained a lot of ground!
![image](https://github.com/arendst/Tasmota/assets/8540074/dcb870b9-f32f-4ba8-acbb-aa0a57582abb)
I can now turn them off and on at the PWM level set on the sliders from Home Assistant.  But, I have no idea how to set a PWM value so that I can make the bright during the day and dim at night.  The PWM values are fairly balanced.

![image](https://github.com/arendst/Tasmota/assets/8540074/655ad6ad-c35a-45f9-8f2f-0d9463710c37)

**Comment by mattlward** [comment]
I just realize that I was making this more complicated than I needed to!  The lights were imported by mqtt and brightness just works!
